docs/dev/guide/custom-aa-tutorial.md

```diff
@@ -0,0 +1,637 @@
+        }
+
+        // Needed for the transaction to be correctly processed by the server.
+        require(success);
+	}
+
+	function isValidSignature(bytes32 _hash, bytes calldata _signature) public override view returns (bytes4) {
+        // The signature is the concatenation of the ECDSA signatures of the owners
+        // Each ECDSA signature is 65 bytes long. That means that the combined signature is 130 bytes long. 
+        require(_signature.length == 130, 'Signature length is incorrect');
+
+        address recoveredAddr1 = ECDSA.recover(_hash, _signature[0:65]);
+        address recoveredAddr2 = ECDSA.recover(_hash, _signature[65:130]);
+
+        require(recoveredAddr1 == owner1);
+        require(recoveredAddr2 == owner2);
+
+        return EIP1271_SUCCESS_RETURN_VALUE;
+	}
+
+	receive() external payable {
+        // If the bootloader called the `receive` function, it likely means
+        // that something went wrong and the transaction should be aborted. The bootloader should
+        // only interact through the `validateTransaction`/`executeTransaction` methods.
+        assert(msg.sender != BOOTLOADER_FORMAL_ADDRESS);
+    }
+}
+```
+
+## The factory
+
+Now, let's build a factory that can deploy these accounts. Note, that if we want to deploy AA, we need to interact directly with the `DEPLOYER_SYSTEM_CONTRACT`. For deterministic addresses, we will use `create2AA` method.
+
+The code will look the following way:
+
+```solidity
+// SPDX-License-Identifier: MIT
+
+import '@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol';
+
+contract AAFactory {
+    bytes32 public aaBytecodeHash;
+    constructor(bytes32 _aaBytecodeHash) {
+        aaBytecodeHash = _aaBytecodeHash;
+    }
+
+    function deployAccount(
+        bytes32 salt,
+        address owner1,
+        address owner2
+    ) external returns (address) {
+        return DEPLOYER_SYSTEM_CONTRACT.create2AA(salt, aaBytecodeHash, 0, abi.encode(owner1, owner2));
+    }
+}
+```
+
+Note, that on zkSync, the deployment is not done via bytecode, but via bytecode hash. The bytecode itself is passed to the operator via `factoryDeps` field. Note, that the `_aaBytecodeHash` must be formed in a special way:
+
+- Firstly, it is hashed with sha256
+- Then, the first two bytes are replaced with the length of the bytecode in 32-byte words.
+
+You don't need to worry about it, since our SDK provides a built-in method to do it, that will be explained below.
+
+## Deploying the factory
+
+To deploy a factory, we need to create a deployment script. Create the `deploy` folder and create one file there: `deploy-factory.ts`. Put the following deployment script there:
+
+```ts
+import { utils, Wallet } from "zksync-web3";
+import * as ethers from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+
+// An example of a deploy script that will deploy and call a simple contract.
```

Comment:
```suggestion
// An example of a deploy script that deploys and call a simple contract

Reply:
I would say `deploys and calls`. Why remove the `.` at the end of the comment? Most of our comments end with a period